test/box-luatest/gh_9923_invalid_iproto_index_name_mapping_on_the_server_side_test.lua

```diff
@@ -0,0 +1,45 @@
+local server = require('luatest.server')
+local t = require('luatest')
+
+local g = t.group()
+
+g.before_all(function()
+    g.server = server:new({alias = 'master'})
+    g.server:start()
+end)
+
+g.after_all(function()
+    g.server:stop()
+end)
+
+g.test_index_by_name = function()
+    g.server:exec(function()
+        local s = box.schema.space.create('test', {
+            if_not_exists = true,
+            field_count = 2,
+            format = {
+                {name = "id", type = "unsigned"},
+                {name = "name", type = "string"}
+            }
+        })
+
+        s:create_index('primary', {
+            type = 'tree',
+            unique = true,
+            parts = {1, 'unsigned'}
+        })
+
+        s:create_index('secondary', {
+            type = 'tree',
+            unique = false,
+            parts = {2, 'string'}
+        })
+
+        local idx_name_result = s.index.secondary:select()
+        t.assert_equals(#idx_name_result, 0)
+
+        local idx_id_result = s.index.primary:select()
+        t.assert_equals(#idx_id_result, 0)
+
+    end)
+end
```

Comment:
The test doesn't cover anything - it works fine on master. Please, check that the new tests actually fail before the fix is applied.

Also:
1. The commit fixes IPROTO resolver - here you use indexes locally.
2. The problem was caused by sparse index id. Here are no spare index ids.